llvm/test/CodeGen/X86/legalize-vec-assertzext.ll

```diff
@@ -34,6 +34,24 @@ define i64 @widen_assertzext(ptr %x) nounwind {
   ret i64 %d
 }
 
+define i64 @widen_assertzext_range_attr(ptr %x) nounwind {
+; CHECK-LABEL: widen_assertzext_range_attr:
+; CHECK:       # %bb.0:
+; CHECK-NEXT:    pushq %rax
+; CHECK-NEXT:    callq test2@PLT
+; CHECK-NEXT:    movb $127, %al
+; CHECK-NEXT:    kmovw %eax, %k1
+; CHECK-NEXT:    vpexpandq %zmm0, %zmm0 {%k1} {z}
+; CHECK-NEXT:    vextracti32x4 $3, %zmm0, %xmm0
+; CHECK-NEXT:    vmovq %xmm0, %rax
+; CHECK-NEXT:    popq %rcx
+; CHECK-NEXT:    vzeroupper
+; CHECK-NEXT:    retq
+  %e = call noundef range(i64 0, 2) <7 x i64> @test2()
+  %d = extractelement <7 x i64> %e, i32 6
+  ret i64 %d
+}
+
```

Comment:
Also test a basic scalar case, and the missing noundef case?

Reply:
hmm the test that was added originally added in https://github.com/llvm/llvm-project/commit/2bba779272a23094b3a1ff7b3c56259d51f457df for this seems to have not failed when the noundef check was added in https://github.com/llvm/llvm-project/commit/9deee6bffa9c331f46c68e5dd4cb4abf93dc0716 shall I add noundef to those test and also use the update_llc_test_checks?
and then add the requested tests?

Reply:
Not sure what you mean here, the test was modified when the noundef requirement was added.

But yes, please add a variant with range to something like llvm/test/CodeGen/AArch64/lower-range-metadata-func-call.ll, rather than this test. This test is for a legalization edge case, while what you want to show is that the AssertZExt improves codegen.

Reply:
I was thinking of the llvm/test/CodeGen/AArch64/lower-range-metadata-func-call.ll and llvm/test/CodeGen/AMDGPU/lower-range-metadata-intrinsic-call.ll that do not have a noundef metadata and still passes

Reply:
Still don't get it -- https://github.com/llvm/llvm-project/blob/main/llvm/test/CodeGen/AArch64/lower-range-metadata-func-call.ll does have noundef metadata? Or do you mean that you removed the metadata and the test did not fail?

Reply:
never mind I mixed up all diffs that I hade open sorry for the noise